pkg/apis/v1alpha1/subnet_types.go

```diff
@@ -0,0 +1,59 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type AccessMode string
+
+// SubnetSpec defines the desired state of Subnet.
+type SubnetSpec struct {
+	// Size of Subnet based upon estimated workload count.
+	// Defaults to 64.
+	// +kubebuilder:default:=64
+	IPv4SubnetSize int `json:"ipv4SubnetSize,omitempty"`
+	// Access mode of Subnet, accessible only from within VPC or from outside VPC.
+	// Defaults to private.
+	// +kubebuilder:default:=private
+	// +kubebuilder:validation:Enum=private;public;isolated
```

Comment:
I do not think any use case for isolated subnet for the first release. @jianjuns  could we just use private/public subnet type? although NSX Subnet API supports private/public/isolated.

Reply:
Yes, for manual defined CRD we can remove "isolated" for now.

Reply:
OK, removed `isolated`, thanks.